internal/db/repositories/user_repository.go

```diff
@@ -0,0 +1,155 @@
+// internal/repositories/user_repository.go
+
+package repositories
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/TheDevExperiment/server/internal/db"
+	"github.com/TheDevExperiment/server/internal/log"
+	"github.com/TheDevExperiment/server/internal/utility/funnynamer"
+	"github.com/TheDevExperiment/server/internal/utility/jwt"
+)
+
+type UserRepository struct {
+	collection *mongo.Collection
+}
+type UserModel struct {
+	Id              primitive.ObjectID `bson:"_id,omitempty" `
+	IsGuest         bool               `bson:"isGuest,omitempty"`
+	GuestAuthSecret string             `bson:"guestAuthSecret,omitempty"`
+	CreatedAt       primitive.DateTime `bson:"createdAt,omitempty"`
+	LastModified    primitive.DateTime `bson:"lastModified,omitempty"`
+	DisplayName     string             `bson:"displayName,omitempty"`
+	CountryID       string             `bson:"countryId,omitempty"`
+	CityID          string             `bson:"cityId,omitempty"`
+	Age             string             `bson:"age,omitempty"`
+	Score           _score             `bson:"score,omitempty"`
+	IsActive        bool               `bson:"isActive,omitempty"`
+	DeletionReason  string             `bson:"deletionReason,omitempty"`
+}
+type UserUpdateModel struct {
+	// Id              primitive.ObjectID `bson:"_id,omitempty" `
+	IsGuest         bool               `bson:"isGuest,omitempty"`
+	GuestAuthSecret string             `bson:"guestAuthSecret,omitempty"`
+	LastModified    primitive.DateTime `bson:"lastModified,omitempty"`
+	DisplayName     string             `bson:"displayName,omitempty"`
+	// CountryID       string             `bson:"countryId,omitempty"`
+	// CityID          string             `bson:"cityId,omitempty"`
+	// Age            string `bson:"age,omitempty"`
+	Score          _score `bson:"score,omitempty"`
+	IsActive       bool   `bson:"isActive,omitempty"`
+	DeletionReason string `bson:"deletionReason,omitempty"`
+}
+
+type _score struct {
+	RatingScoreSum int `bson:"ratingScoreSum"`
+	RatingCount    int `bson:"ratingCount"`
+}
+
+func NewUserRepository() *UserRepository {
+	return &UserRepository{db.GetCollection(viper.GetString("mongodb.db_name"), "users")}
```

Comment:
`touch db/constants.go`
over there lets keep all the constants, for ex 
`const CollectionUsers = "users"`